src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1212,6 +1212,46 @@ WHERE c.uuid IN (:uuids)
         return existingUuids;
     }
 
+    /**
+     * Determines if any of the provided {@link Consumer} UUIDs do not exist or do not belong to the provided
+     * {@link Owner}.
+     *
+     * @param consumerUuids
+     *  the consumer UUIDs to check
+     *
+     * @param ownerKey
+     *  the key of the owner that should own the consumers
+     *
+     * @return all of the provided consumer UUIDs that do not exist or do not belong to the provided owner
+     */
+    public Set<String> getNonExistentConsumerUuids(Collection<String> consumerUuids, String ownerKey) {
+        if (consumerUuids == null || consumerUuids.isEmpty() || ownerKey == null || ownerKey.isBlank()) {
+            return new HashSet<>();
+        }
+
+        String jpql = "SELECT c.uuid " +
+            "FROM Consumer c " +
+            "JOIN Owner o on o.id = c.ownerId " +
+            "WHERE c.uuid IN (:uuids) and o.key = :ownerKey";
+
+        Query query = getEntityManager()
+            .createQuery(jpql, String.class)
+            .setParameter("ownerKey", ownerKey);
+
+        Set<String> uniqueConsumerUuids = new HashSet<>(consumerUuids);
+        Set<String> existingUuids = new HashSet<>();
```

Comment:
This is a lot of extra ID/data juggling for something we can get from the database straight away. We should avoid doubly converting our set. Or, at the very least, if we're going to stick with this paradigm, rather than add our results to a new set, we should instead remove matching results from it and then return it straight away.

Reply:
Updated